### DISCUSSION TITLE
How to set popup placement at bottom side of any other control using MAUI XAML

### DISCUSSION THREAD

**Discussion by dhaval5382** [CommunityToolkit/Maui#2025] [New Feature Discussions]
How to set popup placement at bottom side of any other control using MAUI XAML

**Comment by TheCodeTraveler** [comment]
You'll want to use an anchor. Here's a sample from our Sample App: https://github.com/CommunityToolkit/Maui/blob/main/samples/CommunityToolkit.Maui.Sample/Pages/Views/Popup/PopupAnchorPage.xaml

**Reply by dhaval5382** [reply]
Thanks for reply. I have already used this sample but it is not full fill my exact requirement. My exact popup position on below the button not on bottom page.

**Reply by dhaval5382** [reply]
In WPF side, We have used PlacementTarget property on popup control but we can not used placementtarget on maui popup side.
Just for reference
    private void Invoke(Button popup)
    {
           var dgSearchPopup = new DataGridSearchPopup();
        //dgSearchPopup.Anchor = popup;
        dgSearchPopup.VerticalOptions = LayoutAlignment.End;
        dgSearchPopup.HorizontalOptions = LayoutAlignment.Center;
        dgSearchPopup.BindingContext = this;
        Page.ShowPopup(dgSearchPopup);
    }
    I have also used Anchor property but my popup is open on above my button control not open below button. My requirement is exact popup position in below button not bottom of page.

**Reply by dhaval5382** [reply]
Please refer my image 
![PopupImage](https://github.com/user-attachments/assets/6ea5b029-820c-4053-b892-54204b67a7ae)

**Reply by bijington** [reply]
If you follow the command through to the view model: https://github.com/CommunityToolkit/Maui/blob/main/samples/CommunityToolkit.Maui.Sample/ViewModels/Views/Popup/PopupPositionViewModel.cs you should see how the position can be specified

**Reply by dhaval5382** [reply]
Yes i have already do it but my problem is not resolved.

**Reply by bijington** [reply]
> Yes i have already do it but my problem is not resolved.

Can you open an issue showing us what your issue is then please?

**Reply by dhaval5382** [reply]
Thanks for reply. I have already  share all information.
In WPF side, We have used PlacementTarget property on popup control but we can not used placementtarget on maui popup side.
Just for reference
private void Invoke(Button popup)
{
var dgSearchPopup = new DataGridSearchPopup();
//dgSearchPopup.Anchor = popup;
dgSearchPopup.VerticalOptions = LayoutAlignment.End;
dgSearchPopup.HorizontalOptions = LayoutAlignment.Center;
dgSearchPopup.BindingContext = this;
Page.ShowPopup(dgSearchPopup);
}
I have also used Anchor property but my popup is open on above my button control not open below button. My requirement is exact popup position in below button not bottom of page.
![PopupImage](https://github.com/user-attachments/assets/61dd3fda-16cb-4f57-9ac2-cc9bee11911e)

**Reply by bijington** [reply]
There is no PlacementTarget option. You have to follow the approach shown in this example: https://github.com/CommunityToolkit/Maui/blob/main/samples/CommunityToolkit.Maui.Sample/ViewModels/Views/Popup/PopupPositionViewModel.cs

Have you tried this? If so what happened when you tried it?

**Reply by dhaval5382** [reply]
Yes i do it but my problem is not resolved as per requirement. Please refer my below code.
private void Invoke(Button popup)
{
var dgSearchPopup = new DataGridSearchPopup();
//dgSearchPopup.Anchor = popup;
dgSearchPopup.VerticalOptions = LayoutAlignment.End;
dgSearchPopup.HorizontalOptions = LayoutAlignment.Center;
dgSearchPopup.BindingContext = this;
Page.ShowPopup(dgSearchPopup);
}

**Reply by bijington** [reply]
Can you please open an issue showing us what is not working?

**Reply by dhaval5382** [reply]
if possible, can we connect on Microsoft team so i will show my problem and my logic. If yes, please give me our time.

**Reply by bijington** [reply]
We could connect on teams. Or you could just send a video of the issue if easier?

**Comment by dhaval5382** [comment]
If any update please update me. This issue is very important for me

**Comment by stephenquan** [comment]
Hi @dhaval5382, a trick we've been doing with Popup is creating a transparent full-screen popup, and we manage precise control of the children within.
